api/user/service.go

```diff
@@ -5,60 +5,88 @@ import (
+	"gopkg.in/go-playground/validator.v9"
 )
 
 var ErrUnauthorized = errors.New("unauthorized")
 var ErrConflict = errors.New("conflict")
+var ErrBadRequest = errors.New("bad request")
 
 type Service interface {
-	UpdateDataUser(ctx context.Context, name, username, email, currentPassword, newPassword, ID string) ([]InvalidField, error)
+	UpdateDataUser(ctx context.Context, name, username, email, ID string) ([]InvalidField, error)
+	UpdatePasswordUser(ctx context.Context, currentPassword, newPassword, ID string) error
 }
 
 type service struct {
 	store store.Store
 }
 
 const (
-	conflictName  = "This username already exists"
-	conflictEmail = "This email already exists"
+	ConflictUsername = "This username already exists"
+	ConflictEmail    = "This email already exists"
 )
 
 type InvalidField struct {
-	Name    string
-	Message string
-	Kind    string
+	Username string
+	Message  string
+	Kind     string
 }
 
 func NewService(store store.Store) Service {
 	return &service{store}
 }
 
-func (s *service) UpdateDataUser(ctx context.Context, name, username, email, currentPassword, newPassword, ID string) ([]InvalidField, error) {
+func (s *service) UpdateDataUser(ctx context.Context, name, username, email, ID string) ([]InvalidField, error) {
 	var invalidFields []InvalidField
 
-	user, err := s.store.UserGetByID(ctx, ID)
+	_, err := s.store.UserGetByID(ctx, ID)
 
 	if err != nil {
 		return invalidFields, err
 	}
-	if newPassword != "" && user.Password != currentPassword {
-		return invalidFields, ErrUnauthorized
+
+	validate := validator.New()
+
+	if err := validate.Struct(models.User{
+		Username: username,
+		Email:    email,
+		Name:     name,
+	}); err != nil {
+		return invalidFields, ErrBadRequest
 	}
 
-	var checkName, checkEmail bool
+	var checkUsername, checkEmail bool
 
-	user, err = s.store.UserGetByUsername(ctx, username)
+	user, err := s.store.UserGetByUsername(ctx, username)
 	if err == nil && user.ID != ID {
-		checkName = true
-		invalidFields = append(invalidFields, InvalidField{"username", conflictName, "conflict"})
+		checkUsername = true
+		invalidFields = append(invalidFields, InvalidField{"username", ConflictUsername, "conflict"})
 	}
+
 	user, err = s.store.UserGetByEmail(ctx, email)
 	if err == nil && user.ID != ID {
 		checkEmail = true
-		invalidFields = append(invalidFields, InvalidField{"email", conflictEmail, "conflict"})
+		invalidFields = append(invalidFields, InvalidField{"email", ConflictEmail, "conflict"})
 	}
-	if checkName || checkEmail {
+
+	if checkUsername || checkEmail {
 		return invalidFields, ErrConflict
 	}
-	return invalidFields, s.store.UserUpdate(ctx, name, username, email, currentPassword, newPassword, ID)
+
+	return invalidFields, s.store.UserDataUpdate(ctx, name, username, email, ID)
+}
+
+func (s *service) UpdatePasswordUser(ctx context.Context, currentPassword, newPassword, ID string) error {
+	user, err := s.store.UserGetByID(ctx, ID)
+
+	if err != nil {
```

Comment:
```suggestion
	if user, err := s.store.UserGetByID(ctx, ID); err != nil {
```

Reply:
In this case user will be in a local scope, but needs to be used in later check